### PR TITLE
Fix/bankid collect response

### PR DESCRIPTION
**Pull request by kajm**
## Explain the changes you’ve made
Fix faulty api gateway response causing api gateway to default to 502 

## Explain why these changes are made
The MittHelsingborg application thought that status code 502 is time out, but in fact it is a default status code from apigateway. This caused the application to retry the collect lambda unlimited times

## How to test

1. Checkout this branch
2. run `sls deploy` within service folder for `service bankid-api`
3. try the collect lambda from the application

### CODE REVIEW

File: services/bankid-api/src/lambdas/collect.ts [hunk 1]
```diff
@@ -1,113 +1,64 @@
-import to from 'await-to-js';
-
+import { throwError } from '@helsingborg-stad/npm-api-error-handling';
 import { AxiosInstance, AxiosError } from 'axios';
-import { throwError, InternalServerError } from '@helsingborg-stad/npm-api-error-handling';
+import to from 'await-to-js';
```
Comment (dannil76) [inline services/bankid-api/src/lambdas/collect.ts:3]: Should we avoid the use of 'to'?

File: services/bankid-api/src/lambdas/collect.ts [hunk 2]
```diff
@@ -1,113 +1,64 @@
...
-      user: {
-        personalNumber: string;
-      };
+type BankIdStatus = 'pending' | 'complete' | 'failed';
+
+interface BankIdCollectData {
+  status: BankIdStatus;
+  completionData: {
+    user: {
+      personalNumber: string;
```
Comment (dannil76) [inline services/bankid-api/src/lambdas/collect.ts:30]: We should extract nested structure to separate interface?

File: services/bankid-api/src/lambdas/collect.ts [hunk 3]
```diff
@@ -168,3 +121,72 @@ async function sendBankIdCollectRequest(payload: BankIdCollectLambdaRequest) {
...
+    dependencies.sendBankIdCollectRequest({ orderRef })
+  );
+
+  if (bankIdCollectRequestError) {
+    log.error(
+      'Bank Id Collect request error',
+      'context.awsRequestId',
+      'service-bankid-api-collect-001',
+      bankIdCollectRequestError
+    );
```
Comment (dannil76) [inline services/bankid-api/src/lambdas/collect.ts:139]: Tip, use log.writeError instead

File: services/bankid-api/src/lambdas/collect.ts [hunk 4]
```diff
@@ -168,3 +121,72 @@ async function sendBankIdCollectRequest(payload: BankIdCollectLambdaRequest) {
...
+    const [generateAuthorizationCodeError, authorizationCode] = await to(
+      dependencies.generateAuthorizationCode(personalNumber ?? '')
+    );
+    if (generateAuthorizationCodeError) {
+      log.error(
+        'Bank Id Authorization code error',
+        'context.awsRequestId',
+        'service-bankid-api-collect-002',
+        generateAuthorizationCodeError ?? {}
+      );
```
Comment (dannil76) [inline services/bankid-api/src/lambdas/collect.ts:169]: User log.writeError

File: services/bankid-api/src/lambdas/collect.ts [hunk 5]
```diff
@@ -168,3 +121,72 @@ async function sendBankIdCollectRequest(payload: BankIdCollectLambdaRequest) {
...
+    };
+  }
+
+  return response.success(200, {
+    type: 'bankIdCollect',
+    attributes: responseAttributes,
+  });
+}
+
+export const main = log.wrap(async event => {
```
Comment (dannil76) [inline services/bankid-api/src/lambdas/collect.ts:186]: Leave out async before event :)

**Comment by helsingborgsstad** [comment]
Task linked: [CU-3jv1yd7 Bugg: Collect lambda returnerar felaktigt response till API Gateway](https://app.clickup.com/t/3jv1yd7)

**Comment by helsingborgsstad** [comment]
Task linked: [CU-3ftuvax Bugg: möjlig krasch när BankID collect fallerar](https://app.clickup.com/t/3ftuvax)

**Pull request merged**